admin/controllers/helpdesk.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+const moment = require('moment-timezone')
+
+const checkWindowV2Service = require('../services/check-window-v2.service')
+const config = require('../config')
+const pupilRegisterService = require('../services/pupil-register.service')
+const schoolHomeFeatureEligibilityPresenter = require('../helpers/school-home-feature-eligibility-presenter')
+const schoolImpersonationService = require('../services/school-impersonation.service')
+const schoolService = require('../services/school.service')
+const resultPageAvailabilityService = require('../services/results-page-availability.service')
+const ValidationError = require('../lib/validation-error')
+
+const controller = {}
+
+/**
+ * Renders the helpdesk school impersonation page
+ * @param {object} req
+ * @param {object} res
+ * @param {function} next
+ * @param {object} error
+ * @returns {Promise.<void>}
+ */
+controller.getSchoolImpersonation = async (req, res, next, error = null) => {
+  res.locals.pageTitle = 'MTC Helpdesk School Impersonation'
+  schoolImpersonationService.removeImpersonation(req.user)
+  try {
+    res.render('helpdesk/school-impersonation', {
+      formData: req.body,
+      err: error || new ValidationError(),
+      breadcrumbs: req.breadcrumbs()
+    })
+  } catch (error) {
+    next(error)
+  }
+}
+
+/**
+ * Submit helpdesk school impersonation
+ * @param {object} req
+ * @param {object} res
+ * @param {function} next
+ * @returns {Promise.<void>}
+ */
+controller.postSchoolImpersonation = async (req, res, next) => {
+  const { dfeNumber } = req.body
+  try {
+    const outcome = await schoolImpersonationService.validateImpersonationForm(req.user, dfeNumber)
+    if (outcome.hasError && outcome.hasError()) {
+      return controller.getSchoolImpersonation(req, res, next, outcome)
+    }
+    res.redirect('home')
+  } catch (error) {
+    next(error)
+  }
+  return next()
+}
```

Comment:
Reading this I can't see where any work actually gets done.

Reply:
@jon-shipley changed as described on the other comment